activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -189,8 +189,8 @@ def explain(arel, binds = [], options = []) # :nodoc:
       def insert(arel, name = nil, pk = nil, id_value = nil, sequence_name = nil, binds = [], returning: nil)
         sql, binds = to_sql_and_binds(arel, binds)
         value = exec_insert(sql, name, binds, pk, sequence_name, returning: returning)
-        return id_value if id_value
-        returning.nil? ? last_inserted_id(value) : returning_column_values(value)
+        return returning_column_values(value) if !returning.nil?
+        id_value || last_inserted_id(value)
```

Comment:
I thought the potential fix should wrap the `id_value` into an array like `Array(id_value)` since this is what `_create_record` expects. Not saying that the current solution is wrong, I haven't looked close enough

Reply:
Sounds interesting. I'll look into this further.

From what I have though, my fix appears to be passing the tests.

Reply:
Just my 2 cents: I think the question here boils down to "what do we do if `returning` and `id_value` are both passed?" IMO, the way it's been done here makes the most sense – I would personally expect that the method return the values for all columns specified in `returning`, not just the ID on its own in an array. I guess that technically opens up another question though: should the ID in the array when `returning` is passed be `id_value` or just be whatever the database returned? I personally would expect it to just be what the DB returned untouched, as is done here.

Reply:
One issue I can see here is that `insert()` is a public method and technically Rails 7.1 established that `id_value` will take precedence over any other parameter passed and by changing it to favor `returning:` we are technically changing the API which can't be done in a bugfix.

However, we can put it as if Rails 7.1 made a mistake by favoring `id_value` and it supposed to favor `returning:` from the beginning. Favoring `returning:` in Rails 7.1 wouldn't have been a breaking change since existing usages will continue pass `returning: nil` keeping the priority of `id_value` 

So personally I think we can go with this fix! We may want to adjust the documentation for the method though.

Speaking a bit ahead I feel like there is an option for future version of Rails to deprecate passing `id_value` 
It feels excessive to pass `id_value` to this method which is not used for anything other than a return value. 
Consumers can just use the `id_value `argument directly without relying on it to be returned.
But that's a different story

Reply:
I understand your concern about changing the API. Changing the API seems to be necessary in order to fix this issue.

I like your idea of adjusting documentation of this API to make it more clear for users what it's returning.